ocs_ci/ocs/resources/pvc.py

```diff
@@ -129,15 +130,36 @@ def reclaim_policy(self):
 
     def resize_pvc(self, new_size, verify=False):
         """
-        Returns the PVC size pvc_name in namespace
+        Modify the capacity of PVC
+
+        Args:
+            new_size (int): New size of PVC in Gi
+            verify (bool): True to verify the change is reflected on PVC,
+                False otherwise
 
         Returns:
             bool: True if operation succeeded, False otherwise
         """
-        self.data['status']['capacity']['storage'] = f"{new_size}Gi"
-        self.apply(**self.data)
+        patch_param = f'{{"spec": {{"resources": {{"requests": {{"storage": "{new_size}Gi"}}}}}}}}'
+
+        # Modify size of PVC
+        assert self.ocp.patch(resource_name=self.name, params=patch_param), (
+            f"Patch command to modify size of PVC {self.name} has failed."
+        )
+
         if verify:
-            return self.size == new_size
+            for pvc_data in TimeoutSampler(240, 2, self.get):
+                capacity = pvc_data.get('status').get('capacity').get('storage')
```

Comment:
Don't we check from within a pod that we got the size we asked for after resize?